### PR TITLE
MarketCollateralPool Test

### PR DESCRIPTION
**Pull request by perfectmak**
##### Description
Tests for Market Collateral Pool minting, redeeming and settle and close.

##### Checklist

- [x] Linter status: 100% pass
- [x] Changes don't break existing behavior
- [ ] Tests coverage hasn't decreased

##### Refers/Fixes
Fixes: #159
Fixes: #160

### CODE REVIEW

**Comment by rojosnow** [comment]
@perfectmak Is this WIP PR tied to a specific issue? If so, go ahead and connect it to the PR. If not, set the `Estimate` and the current `Milestone`. Thanks!

**Comment by rojosnow** [comment]
I found the issue and connected it!

**Comment by pelsasser** [comment]
@perfectmak - rebase when you have a chance, everything is deploying properly and tests can now be ran.

The first basic minting test works, so hopefully, now your able to move forward without dealing with the complexities before!

**Comment by pelsasser** [comment]
@perfectmak - reviewing and changing removing WIP from title

File: test/MarketCollateralPool.js [hunk 1]
```diff
@@ -21,95 +21,237 @@ contract('MarketCollateralPool', function(accounts) {
...
-    // try {
-    //   await collateralPool.settleAndClose.call(marketContract.address, { from: accounts[0] });
-    // } catch (err) {
-    //   settleAndCloseError = err;
-    // }
-    // assert.ok(settleAndCloseError instanceof Error, 'settleAndClose() did not fail before settlement');
+      // 3. force contract to settlement
+      const settlementPrice = await settleContract();
+  
+      // 4. redeem all shorts on settlement
```
Comment (pelsasser) [inline test/MarketCollateralPool.js]: @perfectmak should we do the same thing here for the long tokens to ensure it works for both sides (long, short) sides of the market.

File: test/MarketCollateralPool.js [hunk 2]
```diff
@@ -21,95 +21,237 @@ contract('MarketCollateralPool', function(accounts) {
   const entryOrderPrice = 33025;
```
Comment (pelsasser) [inline test/MarketCollateralPool.js]: some of these vars are no longer used, can we clean them up?

**Comment by pelsasser** [comment]
Missing tests (I think)

- [ ] mintPositionTokens fails if market contract is already settled
- [ ] mintPositionTokens TokensMinted event is fired and correctly returns values
- [ ] redeemPositionTokens TokensRedeemed event is fired and correctly returns values
- [ ] settleAndClose burns the expected number of position tokens upon success
- [ ] settleAndClose TokensRedeemed event is fired and correctly returns values

I also think we should consider looking at the totalSupply_ of the erc20 short and long tokens to make sure that is being accounted for correctly, but perhaps @eswarasai is tackling that in his tests of `PositionToken.sol`

File: test/MarketCollateralPool.js (hunk 2 above)
Comment (perfectmak) [reply test/MarketCollateralPool.js]: Sure, cleaning those up now.

**Comment by eswarasai** [comment]
@pelsasser @perfectmak -- So the tests in regards with the `PositionToken.sol` are more of the duplicate to that of `CollateralPool.test.js`. It makes sense for me to have all the passing/failing tests in regards with the `PositionToken` to be under `CollateralPool` since it's the caller of the Long/Short Position Tokens. I'll try to tackle few of the above tests on this branch to breakdown the tasks. Thanks!

**Comment by perfectmak** [comment]
@pelsasser I believe all the case should have been checked now.

**Pull request merged**